source/request_response.c

```diff
@@ -220,6 +220,34 @@ int aws_http_request_add_header(struct aws_http_request *request, struct aws_htt
     return AWS_OP_ERR;
 }
 
+int aws_http_request_add_header_array(
+    struct aws_http_request *request,
+    struct aws_http_header *headers,
+    size_t num_headers) {
+
+    AWS_PRECONDITION(request);
+    AWS_PRECONDITION(headers);
+    AWS_PRECONDITION(num_headers > 0);
```

Comment:
trivial/debatable/whocares: We could allow num_headers == 0

Reply:
![](https://i.imgur.com/FCaHgir.jpg)